htlcswitch.go

```diff
@@ -72,13 +73,20 @@ type htlcSwitch struct {
 	// currently have open with that peer.
 	interfaces map[wire.ShaHash][]*link
 
+	// Maps preimage hashes to outpoints. It allows chained settlement
+	hashRtoChan map[[32]byte] *[]wire.OutPoint
+
 	// TODO(roasbeef): msgs for dynamic link quality
 	linkControl chan interface{}
 
 	outgoingPayments chan *htlcPacket
 
 	htlcPlex chan *htlcPacket
 
+	// Channel for notifying external observers, like payment manager
+	// They should never modify obtained packets
+	htlcPlexOut chan *htlcPacket
```

Comment:
Same goes for this field. Although I do think the notion of allowing the `invoiceRegistry` to receive notifications when invoices are settled is a good idea. A streaming version of the `AddInvoice` RPC could be added which allows the caller to receive an async notification once the invoice has been settled. I'd like to see this explored in a future PR.